jenkins/spark-tests.sh

```diff
@@ -23,7 +23,7 @@ nvidia-smi
 
 ARTF_ROOT="$WORKSPACE/jars"
 MVN_GET_CMD="mvn org.apache.maven.plugins:maven-dependency-plugin:2.8:get -B \
-    -Dmaven.repo.local=$WORKSPACE/.m2 \
+    -Dmaven.repo.local=$WORKSPACE/.m2 -Dtransitive=false \
```

Comment:
This fix would be good for our test script `jenkins/spark-tests.sh`, but there should be a problem on the customers' end, as they may add `rapids-4-spark_2.12` in there pom.xml without adding `transitive=false`, them they will get these dependencies error. 
Would it doable to remove the dependency tree  [dependency reduce] from `rapids-4-spark_2.12`, similar to what we did on branch-21.10? https://github.com/NVIDIA/spark-rapids/pull/3930/files#diff-2681b51a40a6a26d3a6f7f839cd810422226193ef273ea02d1b0b93c3a87cfa6R1868

Reply:
Agreed, the dependencies as published seem problematic for users who want to pull in the dist jar for building their RAPIDS accelerated UDFs or custom code that accesses ColumnarRdd.

Reply:
But in this case it's just wrong.  The dependency states that the jar needs the aggregator jar to be provided, but it contains it instead.  What's the point of advertising a bogus dependency?  It clearly causes problems for some use cases, and is misleading at best if someone examines the pom.

Reply:
I disagree: `provided` is the weakest scope that still serves the purpose that the aggregator dependency tree is required for building the dist jar (which it is, especially given that some branches like spark312db are built remotely). The next weakest dependency scope is `runtime` stating that the dependency will be needed only at the execution phase, and using that might be misleading. 

> What's the point of advertising a bogus dependency?
disagree with the characterization as bogus because the aggregator is required for building dist package. According to the transitive dependency rules: if the `application` depends with any scope on `dist` and `dist` depends with `provided` on `aggregator`, the dependency subtree rooted in `aggregator` is specifically not advertised.

Reply:
IMO this violates the spirit of the shading we're doing in two ways.

1. Most shaded jars do not advertise the dependencies they used during builds when they pull in those artifacts.  That's why there's a dependency-reduced pom built by default by the shade plugin.  What we're doing here is effectively a form of shading, so I do not understand why we would want to advertise the details of a dependency that is irrelevant once the artifact is published.  What's the use-case where having this dependency in the published artifact is desired?  Maybe I'm missing that.

2. I do not think we should reference unpublished dependencies in the poms of our published artifacts, even if they are marked provided. The user does not need to provide it to use the artifact, and the fact that this is a build-time only dependency is not clearly conveyed by the pom as-is.

Bottom line is that this is breaking some workflows as it is now.  Sure, we control this particular workflow and can update it as a workaround, but do we really want to explain this all over again to a user that is doing something similar?  I think it's a bit difficult to argue why we need to list a dependency in this pom as provided when it's not actually needed and then didn't bother to publish a dependency-reduced pom to avoid such problems.

Reply:
We have tried manual install-file with dependency-reduced pom https://github.com/NVIDIA/spark-rapids/pull/4043/commits/372bc1387f822f3b9048d866a5be7e3b25770740#diff-2681b51a40a6a26d3a6f7f839cd810422226193ef273ea02d1b0b93c3a87cfa6R375 

The issue with that is a bug in the install-file mojo when executed inside a multi-module build it's picking a random (based  on the build order of the current maven session) instead of the current pom. I question whether it's worth solving a simple problem affecting this edge case because even a small change is more disruptive https://github.com/NVIDIA/spark-rapids/pull/4043#issuecomment-965766850 than the problem it addresses.

Reply:
Updated the PR based on the offline meeting with @jlowe